udata_front/theme/gouvfr/datagouv-components/src/composables/organizations/useOrganizationCertified.ts

```diff
@@ -1,22 +1,19 @@
-import { ref, onMounted, toValue, type MaybeRefOrGetter } from 'vue';
+import { ref, watchEffect, toValue, type MaybeRefOrGetter } from 'vue';
 import { Organization } from '../../types/organizations';
-
-export const PUBLIC_SERVICE = "public-service";
+import { isType, hasBadge, PUBLIC_SERVICE, LOCAL_AUTHORITY } from './useOrganizationType';
 
 export const CERTIFIED = "certified";
 
-export default function useOrganizationCertified(organizationRef: MaybeRefOrGetter<Organization>) {
+export default function useOrganizationCertified(organizationRef: MaybeRefOrGetter<Organization | null>) {
   const organizationCertified = ref(false);
-  const isOrganizationCertified = () => {
+  watchEffect(() => {
     const organization = toValue(organizationRef);
     if(!organization) {
       organizationCertified.value = false;
       return;
     }
-    organizationCertified.value = organization.badges.some(badge => badge.kind === PUBLIC_SERVICE) &&
-      organization.badges.some(badge => badge.kind === CERTIFIED);
-  };
-  onMounted(isOrganizationCertified);
+    organizationCertified.value = hasBadge(organization, CERTIFIED) && (isType(organization, PUBLIC_SERVICE) || isType(organization, LOCAL_AUTHORITY));
+  });
```

Comment:
This should be a `computed()` for me and not a `watchEffect` but maybe I'm missing something about this project (never use custom `use…` helpers in my Vue apps)